pyanaconda/payload/dnf/payload.py

```diff
@@ -1527,7 +1531,13 @@ def update_base_repo(self, fallback=True, checkmount=True):
 
         # Change the default source to CDROM if there is a valid install media.
         # FIXME: Set up the default source earlier.
-        if checkmount and source_type == DNF_DEFAULT_SOURCE_TYPE and find_optical_install_media():
+
+        # Both DNF default source as well as CDN source should be overriden by
+        # on media source when found.
+        # NOTE: On successfull registration, CDN will be set as installation source
+        #       even if CDROM was set before.
+        can_override_source = source_type in (DNF_DEFAULT_SOURCE_TYPE, SOURCE_TYPE_CDN)
+        if checkmount and can_override_source and find_optical_install_media():
```

Comment:
This should be solved differently.

Reply:
Same here.

Reply:
Yup, much cleaner with the config option. :)